vividus-extension-selenium/src/main/java/org/vividus/ui/context/SelfHealingUiContext.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.ui.context;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Proxy;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import org.openqa.selenium.SearchContext;
+import org.openqa.selenium.StaleElementReferenceException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.WrapsElement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SelfHealingUiContext extends UiContext
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(SelfHealingUiContext.class);
+
+    @Override
+    public SearchContext getSearchContext()
+    {
+        SearchContext searchContext = super.getSearchContext();
+        return wrap(searchContext, SearchContext.class, super::getSearchContext);
+    }
+
+    @Override
+    public <T extends SearchContext> T getSearchContext(Class<T> clazz)
+    {
+        return wrap(super.getSearchContext(clazz), clazz, () -> super.getSearchContext(clazz));
+    }
+
+    @SuppressWarnings({ "unchecked", "AvoidHidingCauseException" })
+    private <T extends SearchContext> T wrap(T searchContext, Class<?> clazz, Supplier<T> searchContextGetter)
+    {
+        if (!(searchContext instanceof WebElement) || !clazz.isInterface())
+        {
+            return searchContext;
+        }
+        Set<Class<?>> proxied = new HashSet<>();
+        proxied.add(WrapsElement.class);
+        proxied.add(WebElement.class);
+        proxied.add(clazz);
```

Comment:
Class<?>[] proxied = { WrapsElement.class, WebElement.class, clazz };
mb?

Reply:
It is a set for a reason :-)

Reply:
It is a set for a reason :-)

Reply:
which reason? :)

Reply:
`searchContext.getSearchContext(WebElement.class)`

Reply:
```
repeated interface: org.openqa.selenium.WebElement
java.lang.IllegalArgumentException: repeated interface: org.openqa.selenium.WebElement
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.validateProxyInterfaces(Proxy.java:720)
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.<init>(Proxy.java:646)
	at java.base/java.lang.reflect.Proxy.lambda$getProxyConstructor$1(Proxy.java:439)
	at java.base/jdk.internal.loader.AbstractClassLoaderValue$Memoizer.get(AbstractClassLoaderValue.java:329)
	at java.base/jdk.internal.loader.AbstractClassLoaderValue.computeIfAbsent(AbstractClassLoaderValue.java:205)
	at java.base/java.lang.reflect.Proxy.getProxyConstructor(Proxy.java:437)
	at java.base/java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1028)
```